src/Plugin.Maui.Audio/AudioRecorder/AudioRecorder.macios.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using AVFoundation;
+using Foundation;
+
+namespace Plugin.Maui.Audio;
+
+partial class AudioRecorder : IAudioRecorder
+{
+	public bool CanRecordAudio => AVAudioSession.SharedInstance().InputAvailable;
+
+	public bool IsRecording => recorder.Recording;
+
+	readonly string destinationFilePath;
+	readonly AVAudioRecorder recorder;
+	readonly TaskCompletionSource<bool> finishedRecordingCompletionSource;
+
+	public AudioRecorder()
+	{
+		InitAudioSession();
+
+		destinationFilePath = GetTempFileName();
+		var url = NSUrl.FromFilename(destinationFilePath);
+
+		var settings = NSDictionary.FromObjectsAndKeys(objects, keys);
+
+		recorder = AVAudioRecorder.Create(url, new AudioSettings(settings), out NSError? error) ?? throw new Exception();
+
+		// TODO: need to tidy this up.
+		recorder.FinishedRecording += Recorder_FinishedRecording;
+		finishedRecordingCompletionSource = new TaskCompletionSource<bool>();
+		recorder.PrepareToRecord();
+	}
+
+	static void InitAudioSession()
+	{
+	    var audioSession = AVAudioSession.SharedInstance();
+
+	    var err = audioSession.SetCategory(AVAudioSessionCategory.PlayAndRecord);
+		if (err is not null)
+		{
+			throw new Exception(err.ToString());
+		}
+
+	    err = audioSession.SetActive(true);
+		if (err is not null)
+		{
+			throw new Exception(err.ToString());
+		}
+	}
+
+	string GetTempFileName()
```

Comment:
It seems this gets a full path rather than just a name, so maybe name it like that